delivery_tnt_oca/models/stock_picking.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    tnt_consignment_data = fields.Serialized()
+    consignment_mumber = fields.Char(sparse="tnt_consignment_data")
+    consignment_date = fields.Char(sparse="tnt_consignment_data")
+    consignment_free_circulation = fields.Char(sparse="tnt_consignment_data")
+    consignment_sort_split = fields.Char(sparse="tnt_consignment_data")
+    consignment_destination_depot = fields.Char(sparse="tnt_consignment_data")
+    consignment_destination_depot_day = fields.Integer(sparse="tnt_consignment_data")
+    consignment_cluster_code = fields.Char(sparse="tnt_consignment_data")
+    consignment_origin_depot = fields.Char(sparse="tnt_consignment_data")
+    consignment_product = fields.Char(sparse="tnt_consignment_data")
+    consignment_option = fields.Char(sparse="tnt_consignment_data")
+    consignment_market = fields.Char(sparse="tnt_consignment_data")
+    consignment_transport = fields.Char(sparse="tnt_consignment_data")
+    consignment_transit_depot = fields.Char(sparse="tnt_consignment_data")
+    consignment_xray = fields.Char(sparse="tnt_consignment_data")
+    tnt_piece_data = fields.Serialized()
+    piece_barcode = fields.Char(sparse="tnt_piece_data")
+    tnt_last_request = fields.Text(
+        string="Last TNT API request", help="Used for debugging", readonly=True
+    )
+    tnt_last_response = fields.Text(
+        string="Last TNT API response", help="Used for debugging", readonly=True
+    )
```

Comment:
Again, use logs please

Reply:
The debate on these fields is here: https://github.com/OCA/delivery-carrier/pull/430#discussion_r741658032
In my opinion these fields are quite useful if we want to know (without entering the logs) the request and response obtained for each record.

Reply:
I don't think an end user can interpret api responses properly. Also I don't think having only the *last* request and response data is helpful (because what happens if the one that failed wasn't the last?).

IMHO it's better to use tools for what they are, and for debugging we have the logs. 🤷🏼‍♂️

Reply:
In `delivery_schenker` I use them (https://github.com/OCA/delivery-carrier/blob/bb074e0da12737ad0fcce85357b10d15ef3fe50d/delivery_schenker/models/delivery_carrier.py#L186-L187)

And I must say I prefer them as well. They even remove the overhead of having to trick separate cursors and stuff like that as it comes with the core internals already.